File: vue/src/store/entities/class.ts
import Entity from "./entity";
import Order from "./order";
import Lesson from "./lesson";
import Course from "./course";
import Teacher from "./teacher";

export default class Class extends Entity<number>{
    name?: string   
    orders?: Array<Order>
    lessons?: Array<Lesson>
    courseId?: number
    course?: Course
    teacher?: Teacher
    teacherId?: number
    lastModificationTime?: string
    lastModifierUserId?: number
    creationTime?: string
    creatorUserId?: number
}

File: aspnet-core/src/EducationAdmin.Core/Education/Class.cs
using Abp.Domain.Entities.Auditing;
using EducationAdmin.Authorization.Users;
using EducationAdmin.Sales;
using EducationAdmin.Students;
using System;
using System.Collections.Generic;
using System.Text;

namespace EducationAdmin.Education
{
    public class Class:FullAuditedEntity<long>
    {
        public string Name { get; set; }


        public List<Order>  Orders { get; set; }

        public List<Lesson> Lessons { get; set; }   

        public long CourseId { get; set; }

        public Course Course { get; set; }

        public User Teacher { get; set; }

        public long TeacherId { get; set; }

        public ClassState State { get; set; }

        
    }


    public enum ClassType
    {
        OneToMany=0,
        OnoToOne=1
    }

    public enum ClassState
    {
        Created=0,

        Close=1
    }
}
